core/src/HIP/Kokkos_HIP_Parallel_Team.hpp

```diff
@@ -653,6 +656,35 @@ class ParallelReduce<CombinedFunctorReducerType,
     }
   }
 
+  int compute_block_count() {
```

Comment:
At the very least make it a const member function so the reader knows in a glance that you do not modify data members.
I think that making it a static function and explicitly passing the policy might increase readability further but I will not block on this part.